### ISSUE TITLE
How to use tags to trigger CI builds

### ISSUE THREAD

**Issue by tuomas2** [AndBible/and-bible#779]
I'm tagging in following situations:

- Tagging on release branch, to make test build before releasing to production (not changing necessarily version number)
- Tagging on feature branch, to make test builds (not changing necessarily version number)
- Tagging on master branch, to make test builds (kind of beta builds, but not official releases)
- Making official production releases (already ok, production-*)
- Making official beta releases (already ok, beta-*)

Need to figure out how to handle all these situations.
Let's think about how to do these and how to implement CI to handle these nicely.

**Comment by timbze** [comment]
Maybe a table like this can help us get closer to what we really would like. Let me know your thoughts / or make changes here @tuomas2 

Since you mentioned "not changing necessarily version number", do you mean that in some cases the version number should increment automatically?

| **<br>**                              | **Create release with APK** | **Release title**                                         | **Release body**     |
| ------------------------------------- | --------------------------- | --------------------------------------------------------- | -------------------- |
| build- **(TAG)**                          |                             |                                                           |                      |
| \-any branches                              | Yes                         | Test release {versionName}-{versionCode}, {TAG} on {branch name}                            | Blank or last commit msg  |
| beta- **(TAG)**                           |                             |                                                           |                      |
| \-develop                              | Yes                         | Beta release {versionName}-{versionCode}, {TAG} on {branch name} | Changelog, if exists |
| \-current\-stable                              | Yes ?                         | Beta release {versionName}-{versionCode}, {TAG} on {branch name} | Changelog, if exists |
| \-release/\*                          | Yes ?                        | Beta release {versionName}-{versionCode}, {TAG} on {branch name} | Changelog, if exists |
| \-feature (any other branch?)         | No                          |                                                           |                      |
| production- **(TAG)**                     |                             |                                                           |                      |
| \-current\-stable                              | Yes                         | Release {versionName}-{versionCode}, {TAG} on {branch name}      | Changelog, if exists |
| \-release/\*                          | Yes                         | Release {versionName}-{versionCode}, {TAG} on {branch name}      | Changelog, if exists |
| \-any other branches         | No                          |                                                           |                      |
| debug- **(TAG)** (use a different app ID) |                             |                                                           |                      |
| \-any branches                        | Yes                         | Debug - {TAG} on {branch name}                                     | Last commit msg      |

**Comment by tuomas2** [comment]
> Since you mentioned "not changing necessarily version number", do you mean that in some cases the version number should increment automatically?

Not necessarily. We are already adding github hash to version string that is displayed in help & tips screen. Perhaps possible tag name should be also displayed in some way (perhaps output of `git describe` command).

**Comment by tuomas2** [comment]
> Blank or last commit msg

Last commit message, or if tag is annotated, tag annotation message. This message body should be explicitly set always so that git issue references are turned automatically to references.

**Comment by timbze** [comment]
Related to #730

**Comment by timbze** [comment]
Outdated
